langchain4j-milvus/src/main/java/dev/langchain4j/store/embedding/milvus/MilvusEmbeddingStore.java

```diff
@@ -42,305 +47,329 @@
+    private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> textSegments) {
+        List<InsertParam.Field> fields = new ArrayList<>();
+        fields.add(new InsertParam.Field(ID_FIELD_NAME, ids));
+        fields.add(new InsertParam.Field(TEXT_FIELD_NAME, toScalars(textSegments, ids.size())));
+        fields.add(new InsertParam.Field(METADATA_FIELD_NAME, toMetadataJsons(textSegments, ids.size())));
+        fields.add(new InsertParam.Field(VECTOR_FIELD_NAME, toVectors(embeddings)));
+
+        insert(milvusClient, collectionName, fields);
+        flush(milvusClient, collectionName);
     }
 
-    /**
-     * @param password The password. See details <a href="https://milvus.io/docs/authenticate.md">here</a>.
-     * @return builder
-     */
-    public Builder password(String password) {
-      this.password = password;
-      return this;
+    @Override
+    public void remove(String id) {
+        ensureNotBlank(id, "id");
+        removeAll(singletonList(id));
     }
 
-    /**
-     * @param consistencyLevel The consistency level used by Milvus.
-     *                         Default value: EVENTUALLY.
-     * @return builder
-     */
-    public Builder consistencyLevel(ConsistencyLevelEnum consistencyLevel) {
-      this.consistencyLevel = consistencyLevel;
-      return this;
+    @Override
+    public void removeAll(Collection<String> ids) {
+        ensureNotEmpty(ids, "ids");
+        removeForVector(milvusClient, collectionName, format("%s in %s", ID_FIELD_NAME, toJson(ids).replace("\n", "")));
     }
 
-    /**
-     * @param retrieveEmbeddingsOnSearch During a similarity search in Milvus (when calling findRelevant()),
-     *                                   the embedding itself is not retrieved.
-     *                                   To retrieve the embedding, an additional query is required.
-     *                                   Setting this parameter to "true" will ensure that embedding is retrieved.
-     *                                   Be aware that this will impact the performance of the search.
-     *                                   Default value: false.
-     * @return builder
-     */
-    public Builder retrieveEmbeddingsOnSearch(Boolean retrieveEmbeddingsOnSearch) {
-      this.retrieveEmbeddingsOnSearch = retrieveEmbeddingsOnSearch;
-      return this;
+    @Override
+    public void removeAll(Filter filter) {
+        ensureNotNull(filter, "filter");
+        removeForVector(milvusClient, collectionName, map(filter));
     }
 
-    /**
-     * @param databaseName Milvus name of database.
-     *                     Default value: null. In this case default Milvus database name will be used.
-     * @return builder
-     */
-    public Builder databaseName(String databaseName) {
-      this.databaseName = databaseName;
-      return this;
+    @Override
+    public void removeAll() {
+        removeForVector(milvusClient, collectionName, "id != \"\"");
```

Comment:
what about [dropping the whole collection](https://milvus.io/docs/quickstart.md#Drop-the-collection) (and re-creating it)? Would be nice to compare the perfromance

Reply:
🆗